src/mixer/basetrackplayer.cpp

```diff
@@ -792,6 +805,38 @@ void BaseTrackPlayerImpl::slotSetTrackColor(const mixxx::RgbColor::optional_t& c
     m_pTrackColor->forceSet(trackColorToDouble(color));
 }
 
+void BaseTrackPlayerImpl::slotTrackColorPrev(double v) {
```

Comment:
I think you can merge these two slots, just the
`color = colorPalette.nextColor(color);`
part needs to be conditional
`if (v > 0) { next } else if (v < 0) { prev } `

Reply:
I'd agree with you, but the v parameter is sent from a press button (so 0 if not pressed and positive if pressed). 
So if my understanding of the ControlPushButton is correct, that mean i'd still need to map the controls to two proxy functions (slotTrackColorNext and slotTrackColorPrev) which would call the general function with 1 or -1 arg.
The benefit is that creating another control type (such as a relative potentiometer) would be easier, the downside is that the call tree gets messier.
What do you think ?

Reply:
Or, I could add a lambda expression in the connect() call !

Reply:
Well, I was thinking of
* slotNextColor
* slotPrevColor
* (slotSelectColor for ControlEncoder +-, for examples see `search_history_selector` or `MoveVertical` controls)

Both would do a `v > 0` check and call (new) `selectColor(double)`, either with `1` or `-1`.
That function does a `v != 0` check, has the body of the slots you have now and the `if (v > 0) { next } else if (v < 0) { prev }`.

(new) `track_color_selector` can call that function directly.

What do you think?

Reply:
> Well, I was thinking of
> 
>     * slotNextColor
> 
>     * slotPrevColor
> 
>     * _(slotSelectColor for the `track_color_selector` (a `ControlEncoder`) +-, for examples see `search_history_selector` or `MoveVertical` controls)_
> 
> 
> Both would do a `v > 0` check and call (new) `selectColor(double)`, either with `1` or `-1`. That function does a `v != 0` check, has the body of the slots you have now and the `if (v > 0) { next } else if (v < 0) { prev }`.
> 
> (new) `track_color_selector` can call that function directly.
> 
> What do you think?

Great !
I had started implementing this using lambdas, i pushed the final version of this (compiled & tested)
what do you think ?